src/frontend/src/flows/recovery/confirmSeedPhrase.ts

```diff
@@ -0,0 +1,157 @@
+import { html, TemplateResult } from "lit-html";
+import { asyncReplace } from "lit-html/directives/async-replace.js";
+import { withRef, renderPage } from "../../utils/lit-html";
+import { Chan } from "../../utils/utils";
+import { ref, createRef, Ref } from "lit-html/directives/ref.js";
+import { mainWindow } from "../../components/mainWindow";
+import { I18n } from "../../i18n";
+
+import copyJson from "./confirmSeedPhrase.json";
+
+// A list of words, where "check" indicates if the user needs to double check (re-input) a word
+type Word = { word: string } & (
+  | { check: false }
+  | { check: true; elem: Ref<HTMLInputElement> }
+);
+
+// A list of indices nicely spread over the 25 words (anchor + 24 BIP39)
+export const checkIndices = [3, 7, 12, 13, 15, 17, 18, 20, 22, 24];
+
+const checkWord = (word: Word): boolean =>
+  word.check
+    ? withRef(word.elem, (elem) => elem.value === elem.dataset.expected) ===
```

Comment:
I'm not sure where is `dataset.expected` set. I guess not part of this PR and already done in the code base. We can safely assume it's correctly set right?

Reply:
Ah, this could a be a little more clear then. It's `data-expected` that's set on the element: https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset